### ISSUE TITLE
Pamgen: CMake warnings about PARSE_ARGUMENTS

### ISSUE THREAD

**Issue by ibaned** [trilinos/Trilinos#1415]
The CI build shows 70 warnings like this coming from Pamgen:
```
CMake Warning at cmake/tribits/core/utils/ParseVariableArguments.cmake:46 (MESSAGE):
  PARSE_ARGUMENTS() is deprecated and should not be used.  Instead use
  CMAKE_PARSE_ARGUMENTS()
Call Stack (most recent call first):
  cmake/tribits/core/utils/ParseVariableArguments.cmake:205 (PARSE_ARGUMENTS_DEPRECATED_WARNING)
  packages/pamgen/test/CMakeLists.txt:33 (PARSE_ARGUMENTS)
  packages/pamgen/test/CMakeLists.txt:91 (PAMGEN_ADD_TEST)
```
Maybe `PAMGEN_ADD_TEST` needs to switch to `CMAKE_PARSE_ARGUMENTS`?

mentioning @trilinos/framework because there is no @trilinos/pamgen

**Comment by bartlettroscoe** [comment]
> Maybe PAMGEN_ADD_TEST needs to switch to CMAKE_PARSE_ARGUMENTS?

Looks like I missed those warnings as part of #1378.  Not sure how I did not see these.  I will fix and push.

**Comment by ibaned** [comment]
Thanks @bartlettroscoe !

**Comment by bartlettroscoe** [comment]
I pushed the commit c00a514 last night that should resolve these warnings (I verified locally that I could reproduce them then observed that the warnings went away).  The build Linux-gcc-4.9.3-SERIAL_Release_gcc_4.9.3__DEV already ran today and showed these warnings disappear as shown at:

* https://testing.sandia.gov/cdash/buildSummary.php?buildid=2945232

See how the "Previous Build" configure warnings were "67" and but are "0"?

Closing as complete.
